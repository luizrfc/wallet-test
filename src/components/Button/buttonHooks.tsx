const { Colors } = require("../../styles/themeStyle");

interface IButtonHooks {
    fontColor: (disable: boolean, type: string) => string
}

const useButton = (): IButtonHooks => {
    const fontColor = (disabled: boolean, type: string): string => {
        if (disabled) return Colors.grey;
        switch (type) {
            case "primary":
                return Colors.blueDark;
            case "secondary":
                return Colors.white;
            case "disabled":
                return Colors.grey;
            case "icon":
                return Colors.secondary;
            case "rounded":
                return Colors.white;
            default:
                return Colors.primary;
        }
    };

    return {
        fontColor
    }
}

export default useButton;