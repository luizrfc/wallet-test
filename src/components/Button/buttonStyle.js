import { StyleSheet } from "react-native";
import { Colors } from "../../styles/themeStyle";

const stylesBase = StyleSheet.create({
  default: {
    display: "flex",
    width: "100%",
    paddingHorizontal: 20,
    paddingVertical: 12,
    justifyContent: "center",
    alignItems: "center",
    borderRadius: 12,
    minHeight: 48
  }
});

const stylesButton = StyleSheet.create({
  primary: {
    backgroundColor: Colors.primary,
    color: Colors.blueDark
  },
  secondary: {
    backgroundColor: Colors.secondary,
    color: Colors.white
  },
  disabled: {
    backgroundColor: Colors.greyLight,
    color: Colors.primary
  },
  icon: {
    backgroundColor: "transparent",
    color: Colors.secondary
  },
  rounded: {
    backgroundColor: Colors.secondary,
    borderRadius: 50,
    width: 42,
    height: 42,
    minHeight: 42,
    padding: 0
  }
});

export { stylesBase, stylesButton };
