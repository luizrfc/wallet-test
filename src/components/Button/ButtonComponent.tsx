import React from "react";
import { GestureResponderEvent, TouchableOpacity } from "react-native";
import { Text, ImageSvg } from "..";
import { stylesBase, stylesButton } from "./buttonStyle";
import useButton from "./buttonHooks";

export type IconsList = "add" | "back" | "wallet" | "camera";
export type TypesList = "primary" | "secondary" | "disabled" | "icon" | "rounded";

interface IButtons {
  text?: string;
  type: TypesList;
  disabled?: boolean;
  icon?: IconsList;
  handleClick?: (event: GestureResponderEvent) => void;
  handleIcon?: () => void;
}

const Button = ({
  text,
  type = "primary",
  disabled = false,
  icon,
  handleClick
}: IButtons): React.ReactNode => {

  const { fontColor } = useButton();

  return (
    <TouchableOpacity
      style={[stylesBase.default, stylesButton[type]]}
      activeOpacity={0.85}
      onPress={handleClick}
      disabled={disabled}
    >
      {icon && (
        <ImageSvg testID="icon-btn-id" image={icon} color={fontColor(disabled, type)} />
      )}
      {text && (
        <Text testID="text-btn-id" text={text} color={fontColor(disabled, type)} />
      )}
    </TouchableOpacity>
  );
};

export default Button;
