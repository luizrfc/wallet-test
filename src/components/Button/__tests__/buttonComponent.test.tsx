import React from "react";
import "react-native";
import { fireEvent, render } from '@testing-library/react-native';
import Button, { TypesList } from "../buttonComponent";
import renderer from 'react-test-renderer';

describe("Button", () => {
  const handleClick = jest.fn();

  it("renders text and handles click event", () => {
    const text = "Click me!";
    const type: TypesList = "primary";
    const disabled = false;

    const { getByTestId } = render(
      <Button
        text={text}
        type={type}
        disabled={disabled}
        handleClick={handleClick}
      />
    );

    const textButton = getByTestId("text-btn-id");
    expect(textButton).toBeTruthy();

    fireEvent.press(textButton);
    expect(handleClick).toHaveBeenCalled();
  });

  it('render Button component correctly', () => {
    const component = renderer.create(<Button
      text="BotÃ£o"
      type="primary"
      handleClick={handleClick}
    />).toJSON();
    expect(component).toMatchSnapshot();
  })
});
