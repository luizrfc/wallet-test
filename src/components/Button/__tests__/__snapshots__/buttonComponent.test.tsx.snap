// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render Button component correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#A5FF32",
      "borderRadius": 12,
      "color": "#142995",
      "display": "flex",
      "justifyContent": "center",
      "minHeight": 48,
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 12,
      "width": "100%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#142995",
          "fontFamily": "PTSans-CaptionRegular",
        },
        {
          "fontSize": 14,
          "lineHeight": 16,
        },
      ]
    }
    testID="text-btn-id"
  >
    Bot√£o
  </Text>
</View>
`;
